lib/msal-browser/src/app/ClientApplication.ts

```diff
@@ -45,6 +45,9 @@ export abstract class ClientApplication {
     // Storage interface implementation
     protected readonly browserStorage: BrowserCacheManager;
 
+    // Native Cache in memory storage implementation
+    protected readonly nativeInternalStorage: BrowserCacheManager | undefined;
```

Comment:
Can we rename this to be more general i.e. not specific to native? (`inMemoryStorage` perhaps?) I expect we will want to use this for other things as well.

Reply:
I intentionally named this `native...` as we already use `inMemory` in `BrowserCacheManager`(we call it `internalStorage`).
 
 `BrowserCacheManager` has three types of storage:

- `internalStorage` - An instance of `inMemory` inside the `browserStorage` instance, used for telemetry, authority etc
- `browserStorage` - instance of browser storage, can be of type `inMemory` if the user chooses it
- `temporaryStorage` - Same instance of `internalStorage` already instantiated in `browserStorage` 

By limiting the new type to `native` calls only, I think, will make it clear that a new instance of `inMemory` storage is created along with `browserStorage` only for `native` cache purposes. All other `inMemory` needs can be met within `browserStorage` instance.
  
We can eventually unify all the different internal memory references, may be. However, it is a major refactor I do not want to start now. Let us discuss if you still believe the name change is needed.

Reply:
> > By limiting the new type to native calls only, I think, will make it clear that a new instance of inMemory storage is created along with browserStorage only for native cache purposes.
> 
> I think this is an artificial constraint. There's no reason this object needs to be scoped only to native caching.
> 
> > All other inMemory needs can be met within browserStorage instance.
> 
> The `internalStorage` inside `BrowserCacheManager` is not the same as this. It is limited to just an implementation of getters and setters, rather than an implementation of our specific caching structure (which this is).
> 

 For both these cases, If a user wants to use `internalMemory` for all their token storage, they can specify the `cacheLocation` as `MemoryStorage`. Do we envision any other use case outside of this, to use `browserStorage` (assume `localStorage` or `sessionStorage`) and cannot leverage `internalStorage` within the `browserStorage`?

Reply:
We agreed on creating a `nativeInternalStorage` instance.